test/functional/rpc_blockchain.py

```diff
@@ -146,7 +146,19 @@ def _test_getblockchaininfo(self):
                         'possible': True,
                     },
                 },
-                'active': False}
+                'active': False
+            },
+            'taproot': {
+                'type': 'bip9',
```

Comment:
I think there is a bit of a misconception here: this doesn't, and doesn't intended to, implement mainchain activation. That's still up for discussion, and BIP341 states that activation is TBD.

BIP9 is chosen here for regtest activation as it's easiest in the current codebase, so that the consensus changes can be tested. Mainchain activation will probably follow in a separate PR, possibly in a different version (the consensus changes usually go in a major release, with activation in a later minor release once there is community consensus on that).